src/wallet/rpcdump.cpp

```diff
@@ -1482,3 +1482,263 @@ UniValue importmulti(const JSONRPCRequest& mainRequest)
 
     return response;
 }
+
+static UniValue ProcessDescriptorImport(CWallet * const pwallet, const UniValue& data, const int64_t timestamp) EXCLUSIVE_LOCKS_REQUIRED(pwallet->cs_wallet)
+{
+    UniValue warnings(UniValue::VARR);
+    UniValue result(UniValue::VOBJ);
+
+    try {
+        if (!data.exists("desc")) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Descriptor not found.");
+        }
+
+        const std::string& descriptor = data["desc"].get_str();
+        const bool active = data.exists("active") ? data["active"].get_bool() : true;
+        const bool internal = data.exists("internal") ? data["internal"].get_bool() : false;
+        const std::string& label = data.exists("label") ? data["label"].get_str() : "";
+        const std::string& output_param = data.exists("address_type") ? data["address_type"].get_str() : "";
```

Comment:
I don't think it is necessary for us to get the output type from the user or from the wallet, it should be something that we can infer from the descriptor. After all, the output type from the descriptor does need to match what we think the output type for the ScriptPubKeyMan is.